src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -584,23 +584,16 @@ private void processCloudWatchAttemptLogInformation(Map<String, Set<String>> com
     @SuppressWarnings("PMD.CollapsibleIfStatements")
     private void processLogsAndUpload() throws InterruptedException {
         while (true) {
-            // If there is already an upload ongoing, don't do anything. Wait for the next schedule to trigger to
-            // upload new logs.
+            //TODO: this is only done for passing the current text. But in practise, we don`t need to intentionally
+            // sleep here.
             if (!isCurrentlyUploading.compareAndSet(false, true)) {
-                //TODO: Use a CountDownLatch so that we could await here instead of sleep?
                 TimeUnit.SECONDS.sleep(periodicUpdateIntervalSec);
                 continue;
             }
-            AtomicReference<Optional<ComponentLogFileInformation>> componentLogFileInformation =
-                    new AtomicReference<>(Optional.empty());
-            AtomicBoolean isActiveFileCompleted = new AtomicBoolean(false);
+            List<ComponentLogFileInformation> unitsOfWork = new ArrayList<>();
             // Get the latest known configurations because the componentLogConfigurations can change if a new
             // configuration is received from the customer.
             for (ComponentLogConfiguration componentLogConfiguration : componentLogConfigurations.values()) {
-                if (componentLogFileInformation.get().isPresent()) {
```

Comment:
**Here is the reason why these changes are needed**

The current implementation builds a single `componentLogFileInformation` breaks the for loop on line 596 as soon as it processes 1 item from the `componentLogConfigurations`. This makes some assumptions about the logs and when adding the code for processing the active file.

The current implementation works because we are not processing the active log file so line 607 was returning empty and it would process the next component configuration on the next run of the while loop. When processing the active log file we run into a scenario where the line 607 never returns true causing other component logs to not be processed.

This is just a steps towards the right direction and it will be further refactored but without this uploading the active file for multiple components won't work